app-cpp/app355.cpp

```diff
@@ -481,6 +208,182 @@ void SatOrbit::OnPrintTrains(const std::vector<std::vector<OrbitalData>>& trainV
+    return satOrbit;
+}
+
+ // SatOrbit
+void SatOrbitMulti::OnCalculateOrbitalData(const std::vector<sat355::TLE>& inTLEVector, OrbitalDataVector& ioDataVector)
+{
+        OnCalculateOrbitalDataMulti(inTLEVector.begin(), inTLEVector.end(), ioDataVector);
+}
+
+void SatOrbitMulti::OnSortOrbitalVector(std::vector<OrbitalData>& ioOrbitalVector)
+{
+    // Multithreaded sort
+    OnSortOrbitalVectorMulti(ioOrbitalVector.begin(), ioOrbitalVector.end());
+}
+
+// SatOrbitMulti
+void SatOrbitMulti::OnCalculateOrbitalDataMulti(const tle_const_iterator& tleBegin, const tle_const_iterator& tleEnd, OrbitalDataVector& ioDataVector)
+{
+    const auto size = std::distance(tleBegin, tleEnd);
+    std::vector<OrbitalData> orbitalVector{};
+    orbitalVector.reserve(size);
+    std::for_each(tleBegin, tleEnd, [&](const sat355::TLE& inTLE)
+    {
+        double out_tleage = 0.0;
+        double out_latdegs = 0.0;
+        double out_londegs = 0.0;
+        double out_altkm = 0.0;
+
+        // Update TLE list with web address
+        // https://celestrak.org/NORAD/elements/gp.php?NAME=Starlink&FORMAT=TLE
+        // get current time as a long long in seconds
+        
+        //long long testTime = time(nullptr);
+        long long testTime = 1705781559; // Time TLEs stored in StarlinkTLE.txt were recorded
+
+        int result = orbit_to_lla(testTime, inTLE.GetName().data(), inTLE.GetLine1().data(), inTLE.GetLine2().data(), &out_tleage, &out_latdegs, &out_londegs, &out_altkm);
+        if (result == 0)
+        {
+            // Warning! Cannot use tle anymore as it has been moved!
+            OrbitalData data(inTLE, out_latdegs, out_londegs, out_altkm);
+            orbitalVector.push_back(std::move(data));
+        }
+    });
+
+    auto& [mutex, outputVector] = ioDataVector; // C++17 Structured Binding simplifies tuple unpacking
+    // Use mutex to protect access to the list
+    {
+        std::lock_guard<std::mutex> lock(mutex);
+        outputVector.insert(outputVector.end(), orbitalVector.begin(), orbitalVector.end());
+    }
+}
+
+void SatOrbitMulti::OnSortOrbitalVectorMulti(orbit_iterator& inBegin, orbit_iterator& inEnd)
+{
+    // Sort by mean motion
+    std::sort(inBegin, inEnd, [](const OrbitalData& inLHS, const OrbitalData& inRHS) -> bool
+    {
+        return SortPredicate(inLHS, inRHS);
+    });
+}
+
+void SatOrbitMulti::OnSortMergeVectorMulti(orbit_iterator& ioBegin, orbit_iterator& ioMid, orbit_iterator& ioEnd)
+{
+    std::inplace_merge(ioBegin, ioMid, ioEnd, [](const OrbitalData& inLHS, const OrbitalData& inRHS) -> bool
+    {
+        return SortPredicate(inLHS, inRHS);
+    });
+}
+
+#pragma endregion{}
+
+//----------------------------------------
+#pragma region SatOrbit
+
+// Public Non-Virtual Interface
+std::unique_ptr<SatOrbit> SatOrbit::Make(SatOrbitKind inKind)
+{
+    std::unique_ptr<SatOrbit> result = nullptr;
+    std::size_t coreCount = std::thread::hardware_concurrency();
```

Comment:
touche'! `std::thread::hardware_concurrency()`!
you found it. yes! that's the right function to use! 👍 
(however, be sure to make it `const` if you never change it)